### PR TITLE
More customizable norm for expert weights

### PR DESCRIPTION
**Pull request by snarayan21**
Modified args and router weight normalization to enable norms of different orders.

### CODE REVIEW

File: megablocks/layers/arguments.py [hunk 1]
```diff
@@ -24,7 +24,7 @@ class Arguments:
     moe_num_experts : int = 1
     moe_top_k : int = 1
     moe_capacity_factor : int = 1
-    moe_normalize_expert_weights: bool = False
+    moe_normalize_expert_weights: Union[int, float, bool] = False
```
Comment (tgale96) [inline megablocks/layers/arguments.py:27]: Rather than having `bool`, can we have this be `| None`? Then the logic using it is cleaner as well.

Comment (snarayan21) [reply megablocks/layers/arguments.py]: updated this!

File: megablocks/layers/router.py [hunk 2]
```diff
@@ -56,7 +56,8 @@ def forward(self, x):
         scores = self.layer(x.view(-1, x.shape[-1])).softmax(dim=-1)
         expert_weights, expert_indices = self._top_k(scores)
         if self.args.moe_normalize_expert_weights:
-            expert_weights /= expert_weights.sum(dim=-1, keepdim=True)
+            expert_weights /= torch.norm(expert_weights, p=self.args.moe_normalize_expert_weights,
```
Comment (tgale96) [inline megablocks/layers/router.py:59]: nit: can we start a new line before the first arg instead of after the second?

**Review by tgale96 — APPROVED** [review]

File: megablocks/layers/router.py (hunk 2 above)
Comment (snarayan21) [reply megablocks/layers/router.py]: done!

**Pull request merged**